client/layout/index.js

```diff
@@ -129,6 +144,32 @@ Layout.propTypes = {
 	} ).isRequired,
 };
 
+const Layout = compose(
+	withSelect( ( select, { isEmbedded } ) => {
+		// Embedded pages don't send plugin info to Tracks.
+		if ( isEmbedded ) {
+			return;
+		}
+
+		const {
+			getActivePlugins,
+			getInstalledPlugins,
+			isJetpackConnected,
+		} = select( PLUGINS_STORE_NAME );
+
+		return {
+			activePlugins: getActivePlugins(),
+			isJetpackConnected: isJetpackConnected(),
+			installedPlugins: getInstalledPlugins(),
+		};
+	} )
+)( _Layout );
+
+const HydratedLayout = withPluginsHydration( {
+	...window.wcSettings.plugins,
+	jetpackStatus: window.wcSettings.dataEndpoints.jetpackStatus,
+} )( Layout );
```

Comment:
You can add these two functions at the same time in `compose`.

```js
const Layout = compose(
    withSelect( /* .... */ ),
    withPluginsHydration(  /* .... */ )
)( _Layout );
```
Also, in the [next wp.data refactor](https://github.com/woocommerce/woocommerce-admin/pull/4144/files#diff-f5d6c918125bcd6501cbf9f5309c544c), I'm adopting a safer version of this in case some properties are not defined.

```js
withPluginsHydration( {
	...( window.wcSettings.plugins || {} ),
	jetpackStatus:
		window.wcSettings.dataEndpoints &&
		window.wcSettings.dataEndpoints.jetpackStatus,
} ),
```

Reply:
How about this, so `jetpackStatus` will be a boolean?

```javascript
withPluginsHydration( {
	...( window.wcSettings.plugins || {} ),
	jetpackStatus:
		window.wcSettings.dataEndpoints &&
		window.wcSettings.dataEndpoints.jetpackStatus || false,
} ),
```